doc/reference/networking/net_pkt.rst

```diff
@@ -10,6 +10,355 @@ Packet Management
+it will be possible, indepentantly to overwrite, to call:
+
+.. code-block:: c
+
+    net_pkt_available_buffer(pkt);
+
+Or, if headers space needs to be accounted:
+
+.. code-block:: c
+
+    net_pkt_available_payload_buffer(pkt, proto);
+
+Sometime, it might be necessary to place the cursor at a known position directly
+and thus, instead of reading from or writing to the net_pkt, to just go the
+relevant position. :c:func:`net_pkt_skip` is meant for this. For instance if
+one wants to go after the IP header:
+
+.. code-block:: c
+
+    net_pkt_cursor_init(pkt);
+    net_pkt_skip(pkt, net_pkt_ip_header_len(pkt));
+
+
+Data access
+===========
+
+Though the api seen previously is rather simple, it means always
+copying things to/from the net_pkt buffer. In many occasions, it is
+more relevant to access the information stored in the buffer
+contiguously. Especially with network packets which embed headers.
+
+These headers are, most of the time, a known fixed set of bytes. It is
+then more natural to have a structure representing a certain type of
+header.  In addition to this, if it is known the header size appears
+in a contiguous area of the buffer, it will be way more efficient to
+cast the actual position in the buffer to the type of header. Either
+for reading or writing the fields of such header, accessing it
+directly will save memory.
+
+Net pkt comes with a dedicated API for this, built on top of the
+previously described API. It is able to handle both contiguous and
+non-contiguous access transparently.
+
+The API is made of two functions: :c:func:`net_pkt_get_data` and
+:c:func:`net_pkt_set_data`. Such functions take a net_pkt and a data
+access descriptor. The data access descriptor is a structure
+describing the size of the data and, eventually, a pointer to a
+contiguous area of the same size. There are two macros used to define
+a data access despritor. One if it is not possible to tell if the data
+will be in a contiguous area or not: NET_PKT_DATA_ACCESS_DEFINE().
+And one if it is guaranteed the data will always be in a contiguous
+area: NET_PKT_DATA_ACCESS_CONTIGUOUS_DEFINE().
+
+Let's take the example of IP and UDP. Bot IPv4 and IPv6 headers are
+always found at the beginning of the packet and are small enough to
+fit in a net_buf of 128 bytes (for instance, though 64 bytes could be
+chosen).
+
+.. code-block:: c
+
+    NET_PKT_DATA_ACCESS_CONTIGUOUS_DEFINE(ipv4_access, struct net_ipv4_hdr);
+    struct net_ipv4_hdr *ipv4_hdr;
+
+    ipv4_hdr = (struct net_ipv4_hdr *)net_pkt_get_data(pkt, &ipv4_acess);
+
+It would be the same for struct net_ipv4_hdr. About UDP header now, it
+is likely to happen that it will not be in a contiguous area in IPv6
+for instance so:
+
+.. code-block:: c
+
+    NET_PKT_DATA_ACCESS_DEFINE(udp_access, struct net_udp_hdr);
+    struct net_udp_hdr *udp_hdr;
+
+    udp_hdr = (struct net_udp_hdr *)net_pkt_get_data(pkt, &udp_access);
+
+At this point, the inner cursor of the net_pkt points at the beginning
```

Comment:
The "inner cursor" term is quite vague, should we just say "cursor" here?

Reply:
Agree. I think all references to "inner cursor" should just be "cursor" throughout this doc.